[6.0.100-alpha.1.21064.27] dotnet build -clp:NoSummary failed to build project

**Application Name:**  nuke global tool app
**OS:** Windows 10 RS5
**CPU:** X64
**.NET Build Number:** 6.0.100-alpha.1.21064.27
**DevDiv bug:** [https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1266557](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1266557)

**Verify Scenarios:**
1). Windows 10 RS5 X64 + .NET Core SDK build 6.0.100-alpha.1.21064.27: Fail
2). Windows 10 RS5 X64 + .NET Core SDK build 5.0.101 : Pass
3). Windows 10 RS5 X64 + .NET Core SDK build 3.1.300  Pass

 **Repo Machine Environment Info :**
  C:\h\w\AA5A0904\p>dotnet --info 
  .NET SDK (reflecting any global.json):
  Version:   6.0.100-alpha.1.21064.27
  Commit:    4239e5cd10
 
**Runtime Environment:**
  OS Name:     Windows
  OS Version:  10.0.17763
  OS Platform: Windows
  RID:         win10-x64
  Base Path:   C:\Program Files\dotnet\sdk\6.0.100-alpha.1.21064.27\
 
**Host (useful for support):**
   Version: 6.0.0-alpha.1.21063.13
   Commit:  6ded57b3d2
 
**.NET SDKs installed:**
   6.0.100-alpha.1.21064.27 [C:\Program Files\dotnet\sdk]
 
**.NET runtimes installed:**
   Microsoft.AspNetCore.App 6.0.0-alpha.1.21063.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
   Microsoft.NETCore.App 6.0.0-alpha.1.21063.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
   Microsoft.WindowsDesktop.App 6.0.0-alpha.1.21062.4 

**Minimal Repro Steps**
1) dotnet new console
2) dotnet build /nodeReuse:false /p:UseSharedCompilation=false -nologo -clp:NoSummary

**Repro steps for NUKE:**
1) On CMD, install app with : dotnet tool install -g Nuke.GlobalTool
2) Go to an empty directory, type nuke
3)Input Y for every question ( there will be 4)
4)Run : nuke restore

**Expected Result:**
Restore success

**Actual Result:**
C:\Users\v-altunc\.nuget\packages\refit\5.0.23\buildTransitive\netstandard2.0\refit.targets(30,5): error : The filename, directory name, or volume label syntax is incorrect. : 'C:\Users\v-altunc\Desktop\aaad\y\obj\lp:NoSummary\RefitStubs.g.cs' [C:\Users\v-altunc\Desktop\aaad\y\y.csproj]
C:\Program Files\dotnet\sdk\6.0.100-alpha.1.21064.27\Microsoft.Common.CurrentVersion.targets(5472,5): error MSB3191: Unable to create directory "obj\lp:NoSummary\". The directory name is invalid. : 'C:\Users\v-altunc\Desktop\aaad\y\obj\lp:NoSummary\' [C:\Users\v-altunc\Desktop\aaad\y\y.csproj]
    6 Warning(s)
    2 Error(s)

**Findings:**
We noticed, it calls the following command after restore :
ExecSafe { & $env:DOTNET_EXE build $BuildProjectFile /nodeReuse:false /p:UseSharedCompilation=false -nologo -clp:NoSummary --verbosity quiet }
For .net6 SDK,  -clp:NoSummary command cause this issue, same can be repro by building a simple project with the above command, it will build fine on .net5 or 3
`dotnet build /clp:NoSummary` works for both .NET 5 & .NET 6. 
The issue here is only about -clp:NoSummary which does not work for .NET 6.

@dotnet-actwx-bot 
@dotnet/compat

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -26,7 +26,7 @@
     <NETStandardLibraryNETFrameworkVersion>2.0.1-servicing-26011-01</NETStandardLibraryNETFrameworkVersion>
     <NewtonsoftJsonPackageVersion>11.0.1</NewtonsoftJsonPackageVersion>
     <NewtonsoftJsonVersion>$(NewtonsoftJsonPackageVersion)</NewtonsoftJsonVersion>
-    <SystemCommandLineVersion>2.0.0-beta1.20574.7</SystemCommandLineVersion>
+    <SystemCommandLineVersion>2.0.0-beta1.21079.2</SystemCommandLineVersion>
     <SystemDiagnosticsFileVersionInfoVersion>4.0.0</SystemDiagnosticsFileVersionInfoVersion>
     <SystemReflectionMetadataVersion>5.0.0</SystemReflectionMetadataVersion>
     <MicrosoftDotNetSignToolVersion>6.0.0-beta.21076.1</MicrosoftDotNetSignToolVersion>

--- a/src/Cli/dotnet/Parser.cs
+++ b/src/Cli/dotnet/Parser.cs
@@ -88,6 +88,12 @@ namespace Microsoft.DotNet.Cli
             return rootCommand;
         }
 
+        private static CommandLineBuilder DisablePosixBinding(this CommandLineBuilder builder)
+        {
+            builder.EnablePosixBundling = false;
+            return builder;
+        }
+
         public static System.CommandLine.Parsing.Parser Instance { get; } = new CommandLineBuilder(ConfigureCommandLine(RootCommand))
             .UseExceptionHandler(ExceptionHandler)
             .UseHelp()
@@ -95,6 +101,7 @@ namespace Microsoft.DotNet.Cli
             .UseValidationMessages(new CommandLineValidationMessages())
             .UseParseDirective()
             .UseSuggestDirective()
+            .DisablePosixBinding()
             .Build();
 
         private static void ExceptionHandler(Exception exception, InvocationContext context)
@@ -118,7 +125,7 @@ namespace Microsoft.DotNet.Cli
                 context.Console.Error.WriteLine(exception.ToString());
             }
             context.ParseResult.ShowHelp();
-            context.ResultCode = 1;
+            context.ExitCode = 1;
         }
 
         internal class CommandLineConsole : IConsole

--- a/src/Tests/dotnet.Tests/ParserTests/MSBuildArgumentCommandLineParserTests.cs
+++ b/src/Tests/dotnet.Tests/ParserTests/MSBuildArgumentCommandLineParserTests.cs
@@ -1,0 +1,39 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.DotNet.Tools;
+using Microsoft.DotNet.Tools.Build;
+using Microsoft.DotNet.Tools.Publish;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Tests.CommandLineParserTests
+{
+    public class MSBuildArgumentCommandLineParserTests
+    {
+        private readonly ITestOutputHelper output;
+
+        public MSBuildArgumentCommandLineParserTests(ITestOutputHelper output)
+        {
+            this.output = output;
+        }
+
+        [Theory]
+        [InlineData(new string[] { "-property:prop1=true", "-p:prop2=false" }, true)]
+        [InlineData(new string[] { "-property:prop1=true", "-p:prop2=false" }, false)]
+        [InlineData(new string[] { "-detailedSummary" }, true)]
+        [InlineData(new string[] { "-clp:NoSummary" }, true)]
+        [InlineData(new string[] { "-orc" }, true)]
+        [InlineData(new string[] { "-orc" }, false)]
+        public void MSBuildArgumentsAreForwardedCorrectly(string[] arguments, bool buildCommand)
+        {
+            RestoringCommand command = buildCommand ? 
+                BuildCommand.FromArgs(arguments) : 
+                PublishCommand.FromArgs(arguments);
+            command.GetProcessStartInfo().Arguments.Split(' ')
+                .Should()
+                .Contain(arguments);
+        }
+    }
+}

